enhancements/ingress/haproxy-healthcheckinterval.md

```diff
@@ -0,0 +1,291 @@
+the `check inter` duration value may come from (evaluated in order):
+- the `router.openshift.io/haproxy.health.check.interval` annotation on the route
+- a (currently hidden) `ROUTER_BACKEND_CHECK_INTERVAL` environment variable
+- 5 seconds as a default if the previous two values are unset
+
+Note that this `check inter` value also serves as a timeout for health checks sent to servers
+if `timeout check` is not set, but `timeout check` is currently hard-coded to 5 seconds in the OpenShift HAProxy
+configuration.  The minimum value for `check inter` will be 5 seconds.
+(See the Risks and Mitigations section for further discussion.)
+
+`check inter` is set per server for the same backends as `timeout check`, with some
+additional constraints.  For plain HTTP backends, or backends with TLS termination set to edge, re-encrypt, or
+passthrough, it is set only if the endpoint is not idled, and the route has more than one active endpoint.
+
+This proposal will add a new optional parameter, `HealthCheckInterval` to the `IngressControllerTuningOptions` struct,
+to set the duration used in `check inter` at the backend scope for every backend server that currently sets a healthcheck
+interval.
+
+### User Stories
+
+#### As a cluster administrator, I need to reduce the frequency of healthchecks to 20 seconds on all backends
+
+Edit the `IngressController` specification to add `healthCheckInterval` to `tuningOptions`.  The duration of the
+`healthCheckInterval` must be set as a string representing time values.  The time value format is an integer optionally
+followed by a time unit (e.g. "ms", "s", "m", etc.). If no unit is specified, the value is measured in
+milliseconds. More information on the time format can be found in the
+[HAProxy documentation](https://github.com/haproxy/haproxy/blob/v2.2.0/doc/configuration.txt).
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  tuningOptions:
+    healthCheckInterval: "20s"
+...
+```
+
+#### As a cluster administrator, I need to reduce the frequency of healthchecks to 20 seconds on all backends except one
+
+Edit the `IngressController` specification to add `healthCheckInterval` to `tuningOptions` as shown above.  Then set
+the route annotation on the exception route to the default 5 seconds:
+
+```yaml
+apiVersion: v1
+kind: Route
+metadata:
+  annotations:
+    router.openshift.io/haproxy.health.check.interval: "5s"
+...
+```
+
+#### As a cluster administrator, I need to set the frequency of healthchecks to the default
+
+Edit the `IngressController` specification to remove `healthCheckInterval` from `tuningOptions`.  The default is restored.
+
+### API Extensions
+
+This proposal will modify the `IngressController` API by adding a new variable called `HealthCheckInterval` to the
+[`IngressControllerTuningOptions`](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L1193) struct
```

Comment:
Better to use permalinks, or just omit the link (people reading this can figure out which repository to grep).